geo-types/src/macros.rs

```diff
@@ -1,7 +1,9 @@
 /// Creates a [`Point`] from the given coordinates.
 ///
 /// ```txt
-/// point!(x: <number>, y: <number>)
+/// point!(<x_number>, <y_number>)
```

Comment:
Not a blocker from merging if others are okay with it, but I'm personally in favor of requiring the user to specify `x` and `y` in the constructor instead of them having to memorize lng/lat ordering with these unnamed positional arguments. https://macwright.com/lonlat/

Reply:
@frewsxcv I agree with you here, neither am I, but we have lots of those all over the code in the form of `(x,y).into()` (as well as `[x,y]`, and the `point.x_y()` fn, etc.  I think we should either consistently abandon all positional ones, or make them consistently available.

Reply:
It's a good point @frewsxcv .  If we removed the support for two positional args ie `point!(x, y)` in this PR, would a lot break?

Reply:
@rmanoka there are no usages of positional `point!(x,y)` because it doesn't exist. We don't have to introduce it, but there is currently several other functions that take positional x,y arguments, and they need to be switched to this macro in order to simplify z&m migration.

Reply:
So your intention is to add support for point!(x,y,z) and point!(x,y,z,m) as positionals?

An alternative is not have positionals at all, and when migrating to the macros, use named args.  Shall we go with that?

Reply:
I think z and m are far less common, and don't need the short syntaxes. I'm ok to force named args everywhere, but there are quiet a few of the various positional instantions at the moment like .into and ::from. Should we keep those or deprecate them?

Reply:
Yeah, let's track that in another issue, and take a call.

Reply:
thx, removed the positional one for now